libethereum/Client.cpp

```diff
@@ -132,17 +146,32 @@ Client::~Client() {
         cerror << "Instance of SkaleHost was not properly created.";
 
     m_signalled.notify_all();  // to wake up the thread from Client::doWork()
-    stopWorking();
 
     m_tq.HandleDestruction();  // l_sergiy: destroy transaction queue earlier
     m_bq.stop();               // l_sergiy: added to stop block queue processing
 
-    terminate();
+    m_bc.close();
+    std::cout << cc::success( "Blockchain is closed" ) << "\n";
+
+    bool isForcefulExit =
+        ( !m_skaleHost || m_skaleHost->exitedForcefully() == false ) ? false : true;
+    delete_lock_file( m_dbPath );
+    if ( !isForcefulExit )
+        std::cout << cc::success( "Deleted lock file " )
+                  << cc::p( boost::filesystem::canonical( m_dbPath ).string() +
+                            std::string( "/skaled.lock" ) )
+                  << "\n";
+    else
+        std::cout << cc::fatal( "ATTENTION:" ) << " " << cc::error( "Deleted lock file " )
+                  << cc::p( boost::filesystem::canonical( m_dbPath ).string() +
+                            std::string( "/skaled.lock" ) )
+                  << cc::error( " after foreceful exit" ) << "\n";
+    std::cout.flush();
```

Comment:
If write to log (see above) then here we also need to flush log, not cout.

Reply:
This can be removed after resolving problems with shutdown

Reply:
resolved